src/coprocessor/endpoint.rs

```diff
@@ -112,7 +112,10 @@ impl<E: Engine> Endpoint<E> {
         if !req_ctx.context.get_stale_read() {
             self.concurrency_manager.update_max_ts(start_ts);
         }
-        if req_ctx.context.get_isolation_level() == IsolationLevel::Si {
+        if matches!(
+            req_ctx.context.get_isolation_level(),
+            IsolationLevel::Si | IsolationLevel::RcCheckTs
```

Comment:
Can these kinds of conditions be wrapped in a utility function like `is_at_least_si` or `requires_si` so that we will be less likely to forgot to miss updating the condition if we add one more `IsolationLevel` choice in the future?